### PR TITLE
Added CPI configuration

### PR DESCRIPTION
**Pull request by vitoravelino**
screenshot-20180312112905-1162x70
screenshot-20180312112856-1157x192
screenshot-20180312113157-1162x768

### CODE REVIEW

**Comment by mjura** [comment]
[Write user case for cloud provider UI part in Velum](https://trello.com/c/BoNizzYj/364-write-user-case-for-cloud-provider-ui-part-in-velum)

**Comment by MaximilianMeister** [comment]
right now it tries to behave the same as the other cloud providers like ec2 and azure. meaning it expects a dedicated view for that in the worker-bootstrap page. similar to https://github.com/kubic-project/velum/blob/master/app/views/setup/worker_bootstrap_ec2.html.slim

but cant find `/srv/velum/config/data/openstack-types.json` when clicking next.

so i think you need to add an exception to https://github.com/kubic-project/velum/pull/451/files#diff-96d7538446a97258fe291a027a04c6baL57 regarding openstack, to not try to render a cloud worker view

**Comment by vitoravelino** [comment]
> right now it tries to behave the same as the other cloud providers like ec2 and azure. meaning it expects a dedicated view for that in the worker-bootstrap page. similar to https://github.com/kubic-project/velum/blob/master/app/views/setup/worker_bootstrap_ec2.html.slim

You are right. We need to ask public cloud team what we should deliver in this pull request. But I'm aware that not all the options are functional, I was just playing around. I think it will only be EC2 and Azure for now but I also make it ready for openstack too on behind the scenes.

**Comment by vitoravelino** [comment]
I'm wondering if these openstack fields shouldn't be in a dedicated page. I guess when the openstack integration is done, this will be handled. Anyway, only EC2 and Azure are enabled atm.

**Comment by mjura** [comment]
> I'm wondering if these openstack fields shouldn't be in a dedicated page. I guess when the openstack integration is done, this will be handled. Anyway, only EC2 and Azure are enabled atm.

Yes, you are right OpenStack fields are dedicated only for openstack driver, Azure, AWS and GCE have different fields.

**Comment by vitoravelino** [comment]
> Yes, you are right OpenStack fields are dedicated only for openstack driver, Azure, AWS and GCE have different fields.

What I wanted to know if it should be in a dedicated page as it was done for EC2 and Azure or if it's in the same page as shown in the screenshots. Let's wait for some feedback.

File: app/helpers/setup_helper.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# SetupHelper contains all the view helpers for the setup process.
+module SetupHelper
+  def cloud_provider_options
+    [
+      ["Azure", :azure],
+      ["Amazon EC2", :ec2]
+      # Still missing templates / business logic around it
+      # ["Google Compute Engine", :gce],
+      # ["OpenStack", :openstack]
```
Comment (mjura) [inline app/helpers/setup_helper.rb]: "OpenStack" has to be un-commented, you can comment "Azure" an "Amazon"

**Comment by vitoravelino** [comment]
> Why OpenStack integration is disabled?

Because there's no integration ready for it afaik. The integrations submitted by the public cloud team so far were only AWS and Azure. Is OpenStack our responsibility? I thought that the requirements for this were just adding a simple select with the options based on what had been done. I might be wrong, so guide me here if that's the case.

What does OpenStack integration mean? Only the option available or something else (e.g.: instance types)? Because if we enable OpenStack there it will throw an exception because there's stuff missing that I'm not aware of. That's why I added the guy from public cloud team to also review this.

**Comment by vitoravelino** [comment]
Confusions were cleared. I'm going to proceed with the expected changes.

**Comment by vitoravelino** [comment]
@mjura This is good to be tested again.

**Comment by bear454** [comment]
Public Cloud environments will be automatically detected; our test images already provide appropriate selection of the UI workflow, by injecting the cloud:framework pillar at boot in EC2, Azure. There's no need for a dropdown to select, for example, EC2 as a framework.

**Comment by aosthof** [comment]
I agree with @bear454 . By adding this dropdown an additional (unnecessary) step would get added for the user. Let's remove the dropdown step of the UI unless there's an important reason to keep it.

**Comment by mjura** [comment]
@aosthof @bear454: This change is about enabling Kubernetes Cloud Provider Integration driver, it doesn't have anything common with Public Cloud integration yet. If you would like to get more details please check it out https://kubernetes.io/docs/concepts/cluster-administration/cloud-providers/

**Comment by vitoravelino** [comment]
@mjura I've reverted the rename that I did before. Please review this again.

File: app/views/setup/cloud/_openstack_configuration.html.slim [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    = f.text_field :cloud_openstack_project, value: @cloud_openstack_project, class: "form-control"
+  .form-group
+    = f.label :cloud_openstack_region, "Region name"
+    = f.text_field :cloud_openstack_region, value: @cloud_openstack_region, class: "form-control"
+  .form-group
+    = f.label :cloud_openstack_username, "Username"
+    = f.text_field :cloud_openstack_username, value: @cloud_openstack_username, class: "form-control"
+  .form-group
+    = f.label :cloud_openstack_password, "Password"
+    = f.text_field :cloud_openstack_password, value: @cloud_openstack_password, class: "form-control"
```
Comment (MaximilianMeister) [inline app/views/setup/cloud/_openstack_configuration.html.slim]: this should not be a `text_field`

Comment (mjura) [reply app/views/setup/cloud/_openstack_configuration.html.slim]: Yes, I agree

**Comment by MaximilianMeister** [comment]
@vitoravelino @mjura i have added the cloud provider to the pillars api in a separate commit, otherwise salt wouldn't set the pillars

**Comment by vitoravelino** [comment]
> @vitoravelino @mjura i have added the cloud provider to the pillars api in a separate commit, otherwise salt wouldn't set the pillars

Thanks!

**Pull request merged**